### PR TITLE
add a test demo reshape op in group norm moved to output tensor

### PR DESCRIPTION
**Pull request by liqiangxl**
add a test demo reshape op in group norm moved to output tensor, should generate a persistent kernel and a no-op.
@wujingyue, can you add this test to #2639

### CODE REVIEW

**Comment by wujingyue** [comment]
> add a test demo reshape op in group norm moved to output tensor, should generate a persistent kernel and a no-op. @wujingyue, can you add this test to #2639

Thanks. The test works when I rebase this PR on https://github.com/NVIDIA/Fuser/tree/bug2599. Instead of adding this test to #2639, would it be better to submit this PR separately? I've yet to understand why the test has to be shipped with #2639.

File: tests/cpp/test_gpu_view.cpp [hunk 1]
```diff
@@ -2584,39 +2584,77 @@ INSTANTIATE_TEST_SUITE_P(
...
+  auto cg_outputs = executor_cache.runFusionWithInputs({t0, tw, tb});
   auto seg_groups =
       executor_cache.getMostRecentKernelRuntime()->fusionSegments()->groups();
-  EXPECT_EQ(seg_groups.size(), 2);
+
+  EXPECT_THAT(
+      seg_groups,
+      Contains(HeuristicIs(ScheduleHeuristic::InnerPersistent)).Times(1));
+  EXPECT_THAT(
+      seg_groups, Contains(HeuristicIs(ScheduleHeuristic::NoOp)).Times(1));
```
Comment (wujingyue) [inline tests/cpp/test_gpu_view.cpp:2650]:
```suggestion
  EXPECT_THAT(
      seg_groups,
      UnorderedElementsAre(HeuristicIs(ScheduleHeuristic::InnerPersistent), Heuristic(ScheduleHeuristic::NoOp)));
```

File: tests/cpp/test_gpu_view.cpp [hunk 2]
```diff
@@ -2584,39 +2584,77 @@ INSTANTIATE_TEST_SUITE_P(
...
+
   testValidate(
-      executor_cache.fusion(), cg_outputs, {t0}, {ref}, __LINE__, __FILE__);
+      executor_cache.fusion(),
+      cg_outputs,
+      {t0, tw, tb},
+      {t7},
```
Comment (wujingyue) [inline tests/cpp/test_gpu_view.cpp:2656]: I believe you can remove this line and let testValidate compute the reference output.

File: tests/cpp/test_gpu_view.cpp [hunk 3]
```diff
@@ -2584,39 +2584,77 @@ INSTANTIATE_TEST_SUITE_P(
...
         std::vector<int64_t>{1, 3}));
 
-TEST_F(GpuViewTest, GroupNorm) {
+// GroupNorm with the last reshape moved to output tensors
+// first 3 reshapes are fused in InnerPersistent
+// last reshape is NoOp
+TEST_F(GpuViewTest, GroupNormReshapeMovedToOutput) {
   auto fusion = std::make_unique<Fusion>();
   FusionGuard fg(fusion.get());
   const int64_t N = 2, C = 128, H = 16, W = 16, G = 32;
```
Comment (wujingyue) [inline tests/cpp/test_gpu_view.cpp:2593]: Style nit: https://google.github.io/styleguide/cppguide.html#Variable_Names

```suggestion
  constexpr int64_t n = 2, c = 128, h = 16, w = 16, g = 32;
```

**Review by wujingyue — APPROVED** [review]

**Comment by liqiangxl** [comment]
> > add a test demo reshape op in group norm moved to output tensor, should generate a persistent kernel and a no-op. @wujingyue, can you add this test to #2639
> 
> Thanks. The test works when I rebase this PR on https://github.com/NVIDIA/Fuser/tree/bug2599. Instead of adding this test to #2639, would it be better to submit this PR separately? I've yet to understand why the test has to be shipped with #2639.

Sure. Just want to make sure #2639 won't break this test.

**Pull request merged**